fix: make iag5 / iag5-openbao crash when GATEWAY_CONNECT_HOSTS is empty

## Problem

`make iag5` and `make iag5-openbao` both hardcode `GATEWAY5_CONNECT_HOSTS=` (empty string) in the Makefile to suppress the Gateway Manager connection. However, passing an empty string causes IAG5 to panic at startup:

```
panic: runtime error: index out of range [0] with length 0
gitlab.com/itential/automation-gateway/gateway5/internal/gateway.(*Gateway).runConnect
```

IAG5's `runConnect` function attempts to access the first element of the hosts slice without first checking its length. An empty `GATEWAY_CONNECT_HOSTS` produces an empty slice, triggering the panic. The container enters a crash-restart loop and never becomes healthy.

This was flagged as a known risk in PR #34:

> Note: suppressing the Gateway Manager connection relies on passing an empty `GATEWAY_CONNECT_HOSTS`. This resolves correctly in compose; the runtime effect should be confirmed against a running IAG5 (`docker logs gateway5`).

Testing against IAG5 `5.4.1-amd64` confirms the panic occurs.

## Impact

- `make iag5` is unusable as a standalone target — IAG5 crashes immediately
- `make iag5-openbao` has the same issue

## Fix Options

**Option A (Makefile):** Remove the inline `GATEWAY5_CONNECT_HOSTS=` override from both targets so the value from `.env` is used. This lets users point IAG5 at a real Platform (local or cloud) without pulling in Mongo/Redis via `make up`.

Before:
```makefile
@$(IAG5_ENV) GATEWAY5_CONNECT_HOSTS= docker compose --profile gateway5 up -d
```

After:
```makefile
@$(IAG5_ENV) docker compose --profile gateway5 up -d
```

**Option B (IAG5):** Fix `runConnect` to gracefully handle an empty hosts list instead of panicking — but this is an upstream IAG5 fix, not a dev-stack fix.

Option A is the immediate pragmatic fix for the dev stack. Option B is the correct long-term fix in IAG5 itself.

## Steps to Reproduce

```bash
make iag5
docker logs gateway5
# observe: panic: runtime error: index out of range [0] with length 0
```

Tested with: IAG5 `5.4.1-amd64`

--- a/Makefile
+++ b/Makefile
@@ -91,7 +91,7 @@ _ensure-gateway5-image:
 	}
 
 down: ## Stop all services
-	@docker compose $(PROFILES) down
+	@docker compose --profile full --profile ldap --profile mcp --profile openbao down
 
 logs: ## Follow logs (all services, or: make logs LOG=platform)
 	@docker compose logs -f $(LOG)

--- a/Makefile
+++ b/Makefile
@@ -68,12 +68,12 @@ up: ## Start all services
 
 iag5: _ensure-gateway5-image ## Deploy IAG5 (Automation Gateway 5) standalone, no Platform
 	@./scripts/generate-certificates.sh --quiet
-	@$(IAG5_ENV) GATEWAY5_CONNECT_HOSTS= docker compose --profile gateway5 up -d
+	@$(IAG5_ENV) docker compose --profile gateway5 up -d
 	@$(MAKE) --no-print-directory status
 
 iag5-openbao: _ensure-gateway5-image ## Deploy IAG5 + OpenBao side by side (no wiring)
 	@./scripts/generate-certificates.sh --quiet
-	@$(IAG5_ENV) GATEWAY5_CONNECT_HOSTS= docker compose --profile gateway5 --profile openbao up -d
+	@$(IAG5_ENV) docker compose --profile gateway5 --profile openbao up -d
 	@./scripts/configure-openbao.sh --init-only
 	@$(MAKE) --no-print-directory status
 

--- a/README.md
+++ b/README.md
@@ -316,10 +316,14 @@ make iag5            # IAG5 only
 make iag5-openbao    # IAG5 + OpenBao (side by side, initialized and unsealed)
 ```
 
-Both targets generate certificates first and skip the Gateway Manager connection, so
-IAG5 runs without a Platform to register with. They require only the IAG5 image (no
-encryption key or `.env`); if the image is missing, the target attempts a pull and
-points you to `make login` for AWS ECR access.
+Both targets generate certificates first and bring up IAG5 without a Platform. They
+require only the IAG5 image (no encryption key or `.env`); if the image is missing, the
+target attempts a pull and points you to `make login` for AWS ECR access.
+
+With no Platform running, IAG5 logs recurring Gateway Manager connection retries against
+the default `platform:8080` host. This is expected and harmless: the container stays up
+and the gRPC server listens on `50051`. To point IAG5 at a real Platform (local or
+cloud) instead, set `GATEWAY5_CONNECT_HOSTS` in your `.env`.
 
 `make iag5-openbao` brings up OpenBao alongside IAG5 and initializes it, but does not
 wire IAG5 to consume OpenBao secrets — the two simply run together. Get the OpenBao root
@@ -331,9 +335,9 @@ Tear down with `make down` (covers IAG5 via the `full` profile). To also stop Op
 docker compose --profile openbao down
 ```
 
-> **Note**: Suppressing the Gateway Manager connection relies on passing an empty
-> `GATEWAY5_CONNECT_HOSTS`. Confirm against your IAG5 version if you see registration
-> attempts in `docker logs gateway5`.
+> **Note**: An empty `GATEWAY5_CONNECT_HOSTS` is invalid for IAG5 and causes a startup
+> panic, so the compose default always resolves to a non-empty host (`platform:8080`).
+> Override it via `.env` to target a different Gateway Manager.
 
 ## 🔧 Installing Adapters
 

--- a/README.md
+++ b/README.md
@@ -329,11 +329,7 @@ cloud) instead, set `GATEWAY5_CONNECT_HOSTS` in your `.env`.
 wire IAG5 to consume OpenBao secrets — the two simply run together. Get the OpenBao root
 token from `volumes/openbao/init-keys.json`.
 
-Tear down with `make down` (covers IAG5 via the `full` profile). To also stop OpenBao:
-
-```bash
-docker compose --profile openbao down
-```
+Tear down with `make down`, which stops IAG5 and OpenBao together.
 
 > **Note**: An empty `GATEWAY5_CONNECT_HOSTS` is invalid for IAG5 and causes a startup
 > panic, so the compose default always resolves to a non-empty host (`platform:8080`).

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -311,9 +311,10 @@ services:
       GATEWAY_APPLICATION_WORKING_DIR: "/etc/gateway"
 
       # gw manager connection
-      # default targets Platform's Gateway Manager; the iag5 make targets pass an
-      # empty GATEWAY5_CONNECT_HOSTS to run standalone without a manager
-      GATEWAY_CONNECT_HOSTS: "${GATEWAY5_CONNECT_HOSTS-platform:8080}"
+      # defaults to Platform's Gateway Manager; the standalone iag5 targets reuse
+      # this default and IAG5 retries the connection harmlessly when no Platform is
+      # up. an empty value is invalid for IAG5 (it panics), so :- forces the default
+      GATEWAY_CONNECT_HOSTS: "${GATEWAY5_CONNECT_HOSTS:-platform:8080}"
       GATEWAY_CONNECT_CERTIFICATE_FILE: "/etc/ssl/gateway5/gw-manager.pem"
       GATEWAY_CONNECT_PRIVATE_KEY_FILE: "/etc/ssl/gateway5/gw-manager-key.pem"
       GATEWAY_CONNECT_INSECURE_TLS: "true"

